File: src/usecases/todo.service.ts
import { AddTodoDto } from './addTodo.dto';
import { DoneDto } from './done.dto';
import { TodoListDto } from './todoList.dto';
import { StartDto } from './start.dto';
import { TodoDto } from './todo.dto';
import { TodoTitleDto } from './todoTitle.dto';

export interface TodoService {
  getTodoList(
    userId: number,
    fields?: string[],
    exclude_done_todo?: boolean,
  ): Promise<TodoListDto>;

  addTodo(addTodoDto: AddTodoDto): Promise<TodoDto>;

  updateTitle(id: number, newTitle: string): Promise<TodoTitleDto>;

  startTodo(id: number): Promise<StartDto>;

  done(id: number): Promise<DoneDto>;
}

File: src/main/java/com/todoapp/usecases/service/TodoService.java
package com.todoapp.usecases.service;

import com.todoapp.entities.Todo;
import com.todoapp.usecases.dto.*;

import java.util.Optional;

public interface TodoService {
    TodoListDto getTodoList(boolean exclude_done_todo, Optional<String> fields);

    TodoDto addTodo(AddTodoDto addTodoDto);

    void updateTitle(TodoTitleDto todoTitleDto);

    void start(StartDto startDto);

    void done(DoneDto doneDto);
} 